buildah/Dockerfile

```diff
@@ -16,27 +13,42 @@ RUN \
     libseccomp-devel \
     ostree-devel \
     git \
-    bzip2 \
-    go-md2man \
-    runc \
-    skopeo-containers && \
-  # Install buildah.
-  mkdir ~/buildah && \
-  cd ~/buildah && \
+    bzip2
+
+FROM base as runc
+ARG RUNC_REVISION="master"
+
+RUN yum -y install runc
+RUN mkdir ~/runc && \
+  cd ~/runc && \
   export GOPATH=`pwd` && \
   git clone https://github.com/opencontainers/runc ./src/github.com/opencontainers/runc && \
   cd $GOPATH/src/github.com/opencontainers/runc && \
   git checkout "${RUNC_REVISION}" && \
   make runc && \
-  mv runc /usr/bin/runc && \
+  mv runc /usr/bin/runc
+
+FROM base as buildah
+ARG BUILDAH_REVISION="master"
+RUN yum -y install \
+    go-md2man \
+    runc \
+    skopeo-containers
+RUN mkdir ~/buildah && \
+  cd ~/buildah && \
+  export GOPATH=`pwd` && \
   cd $GOPATH/ && \
   git clone https://github.com/containers/buildah ./src/github.com/containers/buildah && \
   cd $GOPATH/src/github.com/containers/buildah && \
   git checkout "${BUILDAH_REVISION}" && \
   make && \
-  make install && \
-  buildah --help
```

Comment:
Calls like this were canaries to ensure things built correctly. Not sure the best place to put this now, if your avoiding `RUN` in the final image.